graphite/metric_test.go

```diff
@@ -0,0 +1,65 @@
+package graphite
+
+import (
+	"testing"
+	"time"
+)
+
+type event struct {
+	metric string
+	value  float64
+	when   int64
+}
+
+type senderSpy struct {
+	sent []event
+}
+
+func (s *senderSpy) Send(metric string, value float64, when int64) error {
+	s.sent = append(s.sent, event{metric, value, when})
+	return nil
+}
+
+func (s senderSpy) getSent() []event {
+	return s.sent
+}
+
+func TestMetricSent(t *testing.T) {
+	sender := senderSpy{}
+	metric := NewMetric("my.prefix", 1*time.Millisecond, &sender)
+
+	metric.Increment()
+	metric.Increment()
+	metric.Increment()
+
+	time.Sleep(3 * time.Millisecond)
```

Comment:
Regarding https://github.com/cloudfoundry/clock - we were not aware. Unfortunately, there is no example or test in this repo, but the description sounds good. I agree, it would be nice to get rid of the sleep. I will push an update.

Reply:
https://github.com/cloudfoundry/bosh-agent/blob/bdfafc66300b442adecd691464656b8a5b7951e4/mbus/timeout_nats_client_test.go#L22 contains a good example on how to use the fakeClock, if you are interested.

Reply:
Actually, that is great. There are no tests, examples or docu in this repo and my first quick try had some strange results. I will have a closer look. Thanks.